### ISSUE TITLE
test_s3:test_bucket_list_return_data_versioning bug

### ISSUE THREAD

**Issue by YuanyuanZh** [ceph/s3-tests#169]
@zhangsw for this test case : s3tests.functional.test_s3:test_bucket_list_return_data_versioning 
need to changes in **boto/s3/key.py endElement  function** as well, otherwise test case always failed , could u help to check? Thanks
`def endElement(self, name, value, connection):
        if name == 'Key':
            self.name = value
        elif name == 'ETag':
            self.etag = value
        elif name == 'IsLatest':
            if value == 'true':
                self.is_latest = True
            else:
                self.is_latest = False
        elif name == 'LastModified':
            self.last_modified = value
        elif name == 'Size':
            self.size = int(value)
        elif name == 'StorageClass':
            self.storage_class = value
        elif name == 'Owner':
            pass
        elif name == 'VersionId':
            self.version_id = value
        else:
            setattr(self, name, value)`

need to set key's  'Content-Encoding'

**Comment by zhangsw** [comment]
@YuanyuanZh I test it on ceph master branch and it is passed.

**Comment by YuanyuanZh** [comment]
If GET key doesn't return Content-Encoding header, this test will pass , since eq(key.content_encoding, key_data['content_encoding']), both side is None. As I know AWS GET/HEAD key doesn't return Content-Encoding header, if you don't specify the Content-Encoding header when you create the key. **But if you provide Content-Encoding header when you create key, this test will failed.Did you test this case?**

**Comment by zhangsw** [comment]
Yes, you're right! But unfortunately we can't modify the boto... Maybe we should remove the ‘Content-Encoding’.
